Userland/Libraries/LibCrypto/Hash/HKDF.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023, stelar7 <dudedbz@gmail.com>
+ * Copyright (c) 2024, Ben Wiederhake <BenWiederhake.GitHub@gmx.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibCrypto/Authentication/HMAC.h>
+
+namespace Crypto::Hash {
+
+// https://www.rfc-editor.org/rfc/rfc5869#section-2
+template<typename HashT>
+class HKDF {
+public:
+    using HashType = HashT;
+    using DigestType = typename HashType::DigestType;
+    using HMACType = typename Crypto::Authentication::HMAC<HashType>;
+
+    // Note: The output is different for a salt of length zero and an absent salt,
+    // so Optional<ReadonlyBytes> really is the correct type.
+    static ErrorOr<ByteBuffer> derive_key(Optional<ReadonlyBytes> maybe_salt, ReadonlyBytes input_keying_material, ReadonlyBytes info, u32 output_key_length)
+    {
+        if (output_key_length > 255 * DigestType::Size) {
+            return Error::from_string_view("requested output_key_length is too large"sv);
+        }
+        // Note that it feels like we should also refuse to run with output_key_length == 0,
+        // but the spec allows this.
+
+        // https://www.rfc-editor.org/rfc/rfc5869#section-2.1
+        // Note that in the extract step, 'IKM' is used as the HMAC input, not as the HMAC key.
+
+        // salt: optional salt value (a non-secret random value); if not provided, it is set to a string of HashLen zeros.
+        ByteBuffer salt_buffer;
+        auto salt = maybe_salt.value_or_lazy_evaluated([&] {
+            salt_buffer.resize(DigestType::Size, ByteBuffer::ZeroFillNewElements::Yes);
+            return salt_buffer.bytes();
+        });
+        HMACType hmac_salt(salt);
+
+        // https://www.rfc-editor.org/rfc/rfc5869#section-2.2
+        // PRK = HMAC-Hash(salt, IKM)
+        auto prk_digest = hmac_salt.process(input_keying_material);
+        auto prk = prk_digest.bytes();
+        ASSERT(prk.size() == DigestType::Size);
```

Comment:
Did you intentionally use ASSERT here (and elsewhere)? From what I understand, most places use VERIFY(), and ASSERT() is for performance-critical code.

Reply:
Good catch! No, that was a mistake. Now that I know the difference (thanks for the explanation!), this should be `VERIFY` instead.